src/WaveTrack.cpp

```diff
@@ -240,7 +271,32 @@ void WaveTrack::SetPanFromChannelType()
       SetPan( -1.0f );
    else if( mChannel == Track::RightChannel )
       SetPan( 1.0f );
-};
+}
+
+bool WaveTrack::LinkConsistencyCheck()
+{
+   if(!PlayableTrack::LinkConsistencyCheck())
+      return false;
+
+   //no need to check existance of the next track
+   auto next = dynamic_cast<WaveTrack*>(*std::next(GetOwner()->Find(this)));
+   if (next == nullptr)
+   {
+       return false;
+   }
+
+   auto linkType = GetLinkType();
+   if (static_cast<int>(linkType) == 1 || //Comes from old audacity version
+       linkType == LinkType::Aligned) 
+   {
+      auto newLinkType = AreAligned(GetClips(), next->GetClips()) 
```

Comment:
At least this small change might be better:  use SortedClipArray() not GetClips() because clips in a WaveTrack are not necessarily stored in sorted order by start time.

(I don't really like that, and maybe WaveTrack should be rewritten instead to guarantee the invariant of sortedness of clips.  But, it's not so yet.)